celestia/celestia.go

```diff
@@ -74,12 +77,21 @@ func (c *CelestiaDA) Commit(daBlobs []da.Blob) ([]da.Commitment, error) {
 }
 
 // Submit submits the Blobs to Data Availability layer.
-func (c *CelestiaDA) Submit(daBlobs []da.Blob) ([]da.ID, []da.Proof, error) {
+func (c *CelestiaDA) Submit(daBlobs []da.Blob, gasPrice float64) ([]da.ID, []da.Proof, error) {
 	blobs, commitments, err := c.blobsAndCommitments(daBlobs)
 	if err != nil {
 		return nil, nil, err
 	}
-	height, err := c.client.Blob.Submit(c.ctx, blobs, blob.DefaultSubmitOptions())
+	options := blob.DefaultSubmitOptions()
+	if gasPrice != 0 {
```

Comment:
Should we use a `gasPrice < 0` sentinel value here instead? Because 0 is a valid `gasPrice` if validators are willing to accept the tx.

Reply:
what happens when gasPrice is negative?

Reply:
Updated. We default to `DefaultSubmitOptions` when `gasPrice` is negative.